src/components/MediaRecorder/classes/MediaRecorderController.ts

```diff
@@ -0,0 +1,353 @@
+        blob,
+        ...this.transcoderConfig,
+      });
+      if (!finalBlob) return;
+
+      this.recordingUri = URL.createObjectURL(finalBlob);
+      const file = createFileFromBlobs({
+        blobsArray: [finalBlob],
+        fileName: this.generateRecordingTitle(finalBlob.type),
+        mimeType: finalBlob.type,
+      });
+
+      const recording = {
+        $internal: {
+          file,
+          id: nanoid(),
+        },
+        asset_url: this.recordingUri,
+        duration: durationMs / 1000,
+        file_size: finalBlob.size,
+        mime_type: finalBlob.type,
+        title: file.name,
+        type: RecordingAttachmentType.VOICE_RECORDING,
+        waveform_data: resampleWaveformData(
+          this.amplitudeRecorder?.amplitudes.value ?? [],
+          this.amplitudeRecorderConfig.sampleCount,
+        ),
+      };
+      this.signalRecordingReady?.(recording);
+      this.recording.next(recording);
+    };
+
+    const handleError = (e: Error) => {
+      logError(e);
+      this.error.next(e);
+      this.notification.next({
+        text: this.t('An error has occurred during the recording processing'),
+        type: 'error',
+      });
+    };
+
+    if (mimeType.match('audio/webm')) {
+      // The browser does not include duration metadata with the recorded blob
+      fixWebmDuration(initialBlob, durationMs, {
+        logger: () => null,
+      })
+        .then(makeVoiceRecording)
+        .catch(handleError);
+    } else {
+      makeVoiceRecording(initialBlob).catch(handleError);
+    }
+  }
+
+  resetRecordingState() {
+    this.recordedData = [];
+    this.recording.next(undefined);
+    this.recordingState.next(undefined);
+    this.recordedChunkDurations = [];
+    this.startTime = undefined;
+  }
+
+  cleanUp() {
+    this.resetRecordingState();
+    if (this.recordingUri) URL.revokeObjectURL(this.recordingUri);
+    this.amplitudeRecorder?.close();
+    if (this.mediaRecorder) {
+      disposeOfMediaStream(this.mediaRecorder.stream);
+      this.mediaRecorder.removeEventListener('dataavailable', this.handleDataavailableEvent);
+      this.mediaRecorder.removeEventListener('error', this.handleErrorEvent);
+    }
+  }
+
+  async start() {
+    // account for requirement on iOS as per this bug report: https://bugs.webkit.org/show_bug.cgi?id=252303
+    if (!navigator.mediaDevices) {
+      const error = new Error('Media recording is not supported');
+      logError(error);
+      this.error.next(error);
+      this.notification.next({ text: this.t('Error starting recording'), type: 'error' });
+      return;
+    }
+
+    if (!this.permission.state.value) {
+      await this.permission.check();
+    }
+
+    if (this.permission.state.value === 'denied') {
+      logError(new Error('Permission denied'));
+      return;
+    }
+
+    try {
+      const stream = await navigator.mediaDevices.getUserMedia(
+        this.mediaType === 'video' ? { audio: true, video: true } : { audio: true },
```

Comment:
Why do we support/need video recordings?

Reply:
I will throw an error if video MIME type is passed.

Reply:
https://github.com/GetStream/stream-chat-react/pull/2339/commits/3577a09da5384ac83bee2aeefd21e538ecd9b3c9